### PR TITLE
Fix compatibility issues with 2201.8.0

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
# Description
$subject.

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/576

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version:
* Operating System:
* Java SDK: 

# Checklist:

### Security checks
 - [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: ballerina/tests/test.bal [hunk 1]
```diff
@@ -1102,7 +1102,7 @@ function testVendorBillGetOperation() {
         recordType: "vendorBill"
     };
     VendorBill|error output = netsuiteClient->getVendorBillRecord(recordInfo);
-    if (output is Vendor) {
+    if (output is VendorBill) {
```
Comment (ayeshLK) [inline ballerina/tests/test.bal:1105]:
```suggestion
    if output is VendorBill {
```

File: ballerina/vendorBill.bal [hunk 2]
```diff
@@ -259,7 +259,7 @@ isolated function wrapVendorBillElementsWithParentElement(string subElements, st
         </urn:record>`;
 }
 
-isolated function getVendorBillResult(http:Response response) returns Vendor|error {
+isolated function getVendorBillResult(http:Response response) returns VendorBill|error {
     xml payload = check formatPayload(response);
     if (response.statusCode == http:STATUS_OK) {
```
Comment (ayeshLK) [inline ballerina/vendorBill.bal:264]:
```suggestion
    if response.statusCode == http:STATUS_OK { 
```

File: ballerina/tests/test.bal [hunk 3]
```diff
@@ -1102,7 +1102,7 @@ function testVendorBillGetOperation() {
         recordType: "vendorBill"
     };
     VendorBill|error output = netsuiteClient->getVendorBillRecord(recordInfo);
```
Comment (ayeshLK) [inline ballerina/tests/test.bal:1104]: Do we need to check for this error case ? We can just update the method as following ?
```ballerina
function testVendorBillGetOperation() returns error? {
    // other logic
    VendorBill output = check netsuiteClient->getVendorBillRecord(recordInfo);
}
```

File: ballerina/account.bal [hunk 4]
```diff
@@ -118,23 +118,39 @@ isolated function mapNewItemGroupRecordFields(NewItemGroup itemGroup) returns st
...
-isolated function getTranslationOrHierarchyVRecordsInXML(Translation translation) returns string {
+isolated function getTranslationOrHierarchyVRecordsInXML(Translation|ItemGroupHierarchyVersions translationOrVersions) returns string {
     string finalResult = EMPTY_STRING;
-    map<anydata>|error translationMap = translation.cloneWithType(MapAnyData);
-    if translationMap is map<anydata> {
-        string[] keys = translationMap.keys();
-        int position = 0;
-        foreach var item in translation {
+    map<anydata>|error translationMap = translationOrVersions.cloneWithType(MapAnyData);
+    if translationMap is error {
```
Comment (ayeshLK) [inline ballerina/account.bal:124]: Any specific reason to ignore this error ?

Comment (NipunaRanasinghe) [reply ballerina/account.bal:124]: This PR only addresses the compatibility issues for update 8 so didn't alter any of the existing implementation logics. We can consider all the implementation improvements as a separate task IMO. Will be sending a separate PR for that.

File: ballerina/tests/test.bal (hunk 3 above)
Comment (NipunaRanasinghe) [reply ballerina/tests/test.bal:1104]: https://github.com/ballerina-platform/module-ballerinax-netsuite/pull/93#discussion_r1325678188

Comment (ayeshLK) [reply ballerina/tests/test.bal:1104]: Agreed :+1:

**Review by ayeshLK — APPROVED** [review]

**Pull request merged**